plugins/orchestrator-backend/src/service/router.ts

```diff
@@ -299,7 +299,23 @@ function setupInternalRoutes(
         return;
       }
 
+      if (!workflowInfo.inputSchema) {
+        res
+          .status(500)
+          .send(`Couldn't fetch workflow input schema ${workflowId}`);
+        return;
+      }
+
       schema = workflowInfo.inputSchema;
+
+      // Add ref schemas (steps) to `required` array, otherwise the form validation might not behave as expected
+      const refNames = Object.entries(schema.properties ?? {})
+        .filter(([_, v]) => typeof v === 'object' && v.$ref)
```

Comment:
Shouldn't the logic for filtering the objects be the same as in 
[isJsonObject](https://github.com/caponetto/backstage-plugins/pull/66/files#diff-a4d752f178f740ff87b4d1034adff7c423df9511db9ec0a6e64e6ecbbe690d48R3)?
This will return true for arrays

Reply:
This is not the same. I'm filtering properties that are objects and have the $ref key.